app/services/preserved_object_handler_results.rb

```diff
@@ -88,12 +98,32 @@ def result_hash(code, msg_args=nil)
   # result_array = [result1, result2]
   # result1 = {response_code => msg}
   # result2 = {response_code => msg}
-  def log_results
+  def report_results
+    combined_error_messages = []
     result_array.each do |r|
       severity = self.class.logger_severity_level(r.keys.first)
       msg = r.values.first
       Rails.logger.log(severity, msg)
+      if r.key?(PreservedObjectHandlerResults::INVALID_MOAB)
+        WorkflowErrorsReporter.update_workflow(druid, 'moab-valid', r.values.first)
+      elsif REPORT_ERROR_LIST.include?(r.keys.first)
+        combined_error_messages << r
```

Comment:
how about `candidate_workflow_results` instead of `combined_error_messages` ?   We are still dealing with "result" objects, after all.  Or  `candidate_workflow_result_array` ...

Or, don't do it this way.  

You could have a method 'report_errors_to_workflows`

```ruby
def report_errors_to_workflows
  # not quite the right syntax, just the idea
  moab-validation-err-result = result_array.find { |r| r.code == PreservedObjectHandlerResults::INVALID_MOAB }
  WorkflowErrorsReporter.update_workflow(druid, 'moab-valid', moab-validation-err-result.values.first) if moab-validation-err-result

  other_results = result_array.dup - moab-validation-err-result
  err_results = other_results.select_all { |r| r << r if REPORT_ERROR_LIST.include?(r.keys.first) }
  report_combined_errs_to_workflows(err_results, stacktrace)
  result_array
end

def report_combined_errs_to_workflows # new name for "display_message"

end
```